src/main/java/org/mockbukkit/mockbukkit/block/state/AbstractFurnaceStateMock.java

```diff
@@ -179,4 +180,33 @@ public FurnaceInventory getSnapshotInventory()
 		return mock;
 	}
 
+	@Override
+	public boolean equals(Object o)
+	{
+		if (this == o) return true;
+		if (!(o instanceof AbstractFurnaceStateMock that)) return false;
+		if (!super.equals(o)) return false;
+		return burnTime == that.burnTime
+				&& cookTime == that.cookTime
+				&& cookTimeTotal == that.cookTimeTotal
+				&& Double.compare(cookSpeedMultiplier, that.cookSpeedMultiplier) == 0;
+	}
+
+	@Override
+	public int hashCode()
+	{
+		return Objects.hash(super.hashCode(), burnTime, cookTime, cookTimeTotal, cookSpeedMultiplier);
+	}
+
+	@Override
+	public String toString()
+	{
+		return "AbstractFurnaceStateMock{" +
```

Comment:
Won't this give a confusing toString response?
```
SomethingStateMock{<states originating from this class>} SuperStateMock{<states originating from super class>}
```
I would be confused by this, as it would actually seem like it's two separate block states // one SomethingStateMock and one SuperStateMock. This might be fine when printing one block state, would still be confusing from my point of view, but it would get even more confusing when printing multiple states, let's say I write this code:
```java
for(BlockState state : states) {
    System.out.print(state + " ")
}
```
^yes this would become an extremely confusing mess (also bad practise as there's other one line alternatives).

Long story short: You need to have all the components of the super state inside of the brackets of the child state for the return value of the `toString` method call.

Just going to note that another point against this is that the string can be recursively extended, i.e. the super class can also have a super class that is a BlockState. So the behavior of the toString method becomes even more complicated.

Reply:
Yeah, I'm not a huge fan of this pattern either. It's what Intellij _suggested_. I've never really like the `Type{properties}` format that is typical to java because of this kind of awkwardness dealing with inheritance. Surprisingly there's only one existing toString impl in MockBukkit and it sidesteps this issue by using the serialization map so I just went with Intellij's suggestion figuring it was roughly java-standard-sop.

Do you have any opinions on how you want it to look?

1. Could move the last curly to the end
```
SomethingStateMock{<states originating from this class>, SuperStateMock{<states originating from super class>}}
```

2a. Could unwrap `super.toString()` states using fun string manipulation to find the first curly and skip past it (and not emit an extra close curly)
```
SomethingStateMock{<states originating from this class>, <states originating from super class>}
```

2b. Same result as 2a but don't call `super.toString()` and make every concretion responsible for printing all parent properties - increased maintenance but clean impl.

3. Could say forget the Type{} part making `super.toString()` usage simple and uncomplicated.
```
<states originating from this class>, <states originating from super class>
```

4. other?

Reply:
Eh, thought of a much more obvious way to implement no.2 - updating PR